crates/bevy_ecs/src/world/deferred_world.rs

```diff
@@ -778,85 +784,36 @@ impl<'w> DeferredWorld<'w> {
         }
     }
 
-    /// Triggers all event observers for [`ComponentId`] in target.
+    /// Triggers all `event` observers for the given `targets`
     ///
     /// # Safety
-    /// Caller must ensure observers listening for `event_key` can accept ZST pointers
+    /// - Caller must ensure `E` is accessible as the type represented by `event_key`
     #[inline]
-    pub(crate) unsafe fn trigger_observers(
+    pub unsafe fn trigger_raw<'a, E: Event>(
```

Comment:
I notice there are a lot of new calls to this method in this PR, but almost none of them has a safety comment matching this precondition.